### PR TITLE
Add encrypted cache support

### LINKED ISSUE #3136: Encrypted cache support
After some preliminary performance testing, we believe encrypted cache devices will still provide a sizable performance benefit.

### PR DESCRIPTION
**Pull request by jbaublitz**
Closes #3136

### CODE REVIEW

**Comment by mulkieran** [comment]
@jbaublitz Plz rebase when you get the chance, thx.

**Comment by mulkieran** [comment]
For initializing cache after pool has created we decided that it would count as a bug if a client using D-Bus revision 4 or below found that they could create a cache even if the pool was encrypted. We'll have the D-Bus layer pass a boolean to the engine, to indicate which decision it should make.

**Review by mulkieran — CHANGES_REQUESTED** [review]
Also one minor comment about tests...

File: src/engine/strat_engine/pool.rs [hunk 1]
```diff
@@ -1289,7 +1290,7 @@ mod tests {
                 .unwrap();
         }
 
-        pool.init_cache(uuid, name, paths1).unwrap();
+        pool.init_cache(uuid, name, paths1, false).unwrap();
```
[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/pool.rs:1293]: If we want to keep it unchanged, that should be `false`. My goal was to not change the tests at all.

[the remark this answers is not in the document]
Comment (mulkieran) [reply src/engine/strat_engine/pool.rs:1293]: I think it is better to change to what stratisd will be doing from now through the foreseeable future, which is always supporting an encrypted cache. In the case where more ```init_cache``` invocations are put in our tests, ```true``` will always be the correct argument (unless we're doing some backwards compatibility testing in the unit tests, which is highly unlikely).

[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/pool.rs:1293]: Okay that sounds good!

[the remark this answers is not in the document]
Comment (jbaublitz) [reply src/engine/strat_engine/pool.rs]: Resolved.

**Review by mulkieran — APPROVED** [review]

**Pull request merged**